docs-v2/reference/scripts.mdx

```diff
@@ -774,6 +778,33 @@ await nango.batchDelete(githubIssuesToDelete, 'GitHubIssue');
   </ResponseField>
 </Expandable>
 
+### Detect deletions automatically
+
+Automatically detects and marks records as deleted by comparing the current sync execution with the previous one.
+
+```js
+await nango.deleteRecordsFromPreviousExecution('ModelName');
+```
+
+This function should be called at the end of your sync execution, after all records have been saved with `batchSave`. Nango will compare the records saved in the current execution with those from the previous execution and mark any missing records as deleted.
```

Comment:
What happens if `deleteRecordsFromPreviousExecution` is called before some calls to `batchSave` for the same model?

Reply:
It is prevented at compilation time: https://github.com/NangoHQ/nango/blob/master/packages/cli/lib/zeroYaml/compile.ts#L254-L260

If it was happening though, it would mark as deleted some records than potentially still exist.. They would be re-instated by following batchSave operations but during a short time period of time they would show as deleted.